common-custom-user-data-maven-extension/README.md

```diff
@@ -30,7 +44,26 @@ This approach has a number of benefits:
 - Tailor the build scan enhancements to exactly the set of tags, links and custom values you require.
 - Standardize the configuration for connecting to Gradle Enterprise and Gradle Build Cache in your organization, removing the need for each project to specify this configuration.
 
-If your customized extension provides all required Gradle Enterprise configuration, then a consumer project will get all the benefits of Gradle Enterprise simply by applying the extension. The plugin sources provide a placeholder and example code to get you started.
+The project sources provide a good template to get started with your own Maven extension. 
+Refer to the [Javadoc](https://docs.gradle.com/enterprise/maven-extension/api/) for more details on the key types available for use.
+
+#### Providing Gradle Enterprise configuration in your custom Maven extension
+
+Using a custom extension, it is possible to completely remove the need for each project to declare a `gradle-enterprise.xml` file.
+Much of the configuration can be supplied programmatically. See `CustomBuildScanConfig` and `CustomBuildCacheConfig` for examples.
+
+At this time, the Gradle Enterprise server URL and credentials cannot be provided programmatically, and must be specified in a `gradle-enterprise.xml` file.
```

Comment:
Not sure why you mention the credentials here. They only need to be set on the remote build cache - which can be done [programmatically](https://docs.gradle.com/enterprise/maven-extension/api/com/gradle/maven/extension/api/cache/Credentials.html).